e2e/metrics/cluster_test.go

```diff
@@ -1,108 +1,174 @@
-		"fleet_cluster_resources_count_modified":     true,
-		"fleet_cluster_resources_count_notready":     true,
-		"fleet_cluster_resources_count_orphaned":     true,
-		"fleet_cluster_resources_count_ready":        true,
-		"fleet_cluster_resources_count_unknown":      true,
-		"fleet_cluster_resources_count_waitapplied":  true,
-		// The value of cluster.Status.Display.State is empty if no issues are
-		// found and this means no metric is created.
-		"fleet_cluster_state": false,
+func expectMetrics(name string, expectedMetrics MetricsSelector, expectExist bool) error {
+	expectToFindOneMetric := func(expectedMetric, labelName, labelValue string) error {
+		labels := map[string]string{
+			"namespace": "fleet-local",
+			"name":      name,
+		}
+		if labelName != "" && labelValue != "" {
+			labels[labelName] = labelValue
+		}
+		metric, err := et.FindOneMetric(expectedMetric, labels)
+
+		if expectExist && err != nil {
+			return err
+		} else if !expectExist && err == nil {
+			return fmt.Errorf("metric found but not expected: %v", metric)
+		}
+		return nil
 	}
 
-	It("should have metrics for all existing cluster resources", func() {
-		Eventually(func() error {
-			var (
-				clustersOut string
-				err         error
-			)
-			clustersOut, err = env.Kubectl.Get(
-				"-A", "clusters.fleet.cattle.io",
-				"-o", "json",
-			)
-			Expect(err).ToNot(HaveOccurred())
-
-			var existingClusters clusters
-			err = json.Unmarshal([]byte(clustersOut), &existingClusters)
-			Expect(err).ToNot(HaveOccurred())
-
-			et := metrics.NewExporterTest(metricsURL)
-
-			Expect(len(existingClusters)).ToNot(BeZero())
-
-			for _, cluster := range existingClusters {
-				for metricName, expectedExist := range expectedMetricsExist {
-					_, err := et.FindOneMetric(
-						metricName,
-						map[string]string{
-							"name":      cluster.Name,
-							"namespace": cluster.Namespace,
-						},
-					)
-					if expectedExist && err != nil {
+	for expectedMetric, expectedLabels := range expectedMetrics {
+		if len(expectedLabels) > 0 {
+			for label, values := range expectedLabels {
+				for _, value := range values {
+					err := expectToFindOneMetric(expectedMetric, label, value)
+					if err != nil {
 						return err
-					} else if !expectedExist && err == nil {
-						return fmt.Errorf(
-							"expected metric %s not to exist, but it exists",
-							metricName,
-						)
 					}
 				}
 			}
-			return nil
-		}).ShouldNot(HaveOccurred())
-	},
-	)
+		} else {
+			err := expectToFindOneMetric(expectedMetric, "", "")
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+var _ = Describe("Cluster Metrics", Label("cluster"), func() {
+	It("should have metrics for the local cluster resource", func() {
+		eventuallyExpectMetrics("local", expectedMetrics, true)
+	})
+
+	When("the initial cluster object has changed in any way", func() {
+		It(
+			"should not have duplicated metrics after a cluster has been changed",
+			Label("modified"),
```

Comment:
Just curious: what is this label for? Same question about `Label("deleted")` on L139.